test/client_test.exs

```diff
@@ -156,6 +158,96 @@ defmodule ClientTest do
       )
     end
 
+    test "when replies are fragmented" do
+      address = {127, 0, 0, 1}
+      port = 4090
+      socket = make_ref()
+      raw_hl7 = HL7.Examples.wikipedia_sample_hl7()
+      message = HL7.Message.new(raw_hl7)
+      packet = MLLP.Envelope.wrap_message(raw_hl7)
+
+      tcp_reply1 =
+        MLLP.Envelope.wrap_message(
+          "MSH|^~\\&|SuperOE|XYZImgCtr|MegaReg|XYZHospC|20060529090131-0500||ACK^O01|01052901|P|2.5\rMSA|AA|01052901|You win!\r"
+        )
+
+      {ack_frag1, ack_frag2} = String.split_at(tcp_reply1, 50)
+
+      MLLP.TCPMock
+      |> expect(
+        :connect,
+        fn ^address, ^port, [:binary, {:packet, 0}, {:active, false}], 2000 ->
+          {:ok, socket}
+        end
+      )
+      |> expect(:send, fn ^socket, ^packet -> :ok end)
+      |> expect(:recv, fn ^socket, 0, :infinity -> {:ok, ack_frag1} end)
+      |> expect(:recv, fn ^socket, 0, :infinity -> {:ok, ack_frag2} end)
+
+      {:ok, client} = Client.start_link(address, port, tcp: MLLP.TCPMock, use_backoff: true)
+
+      expected_ack = %MLLP.Ack{acknowledgement_code: "AA", text_message: "You win!"}
+
+      assert(
+        {:ok, :application_accept, expected_ack} ==
+          Client.send(client, message)
+      )
+
+      {ack_frag1, ack_frag2} = String.split_at(tcp_reply1, 50)
+
+      {ack_frag2, ack_frag3} = String.split_at(ack_frag2, 10)
+
+      MLLP.TCPMock
+      |> expect(:send, fn ^socket, ^packet -> :ok end)
+      |> expect(:recv, fn ^socket, 0, :infinity -> {:ok, ack_frag1} end)
+      |> expect(:recv, fn ^socket, 0, :infinity -> {:ok, ack_frag2} end)
+      |> expect(:recv, fn ^socket, 0, :infinity -> {:ok, ack_frag3} end)
+
+      assert(
+        {:ok, :application_accept, expected_ack} ==
+          Client.send(client, message)
+      )
+    end
+
+    test "when replies are fragmented and the last fragment is not received" do
```

Comment:
This is unfortunately a brittle test. If we keep this approach, then we might want to tag it as such and not run it by default.